arrow/src/test/java/org/apache/iceberg/arrow/vectorized/ArrowReaderTest.java

```diff
@@ -0,0 +1,734 @@
+    readAndCheckVectorSchemaRoots(scan, NUM_ROWS_PER_MONTH, NUM_ROWS_PER_MONTH, ALL_COLUMNS);
+  }
+
+  /**
+   * Read selected rows and all columns from the table using a time range row filter.
+   * The test asserts that the result is empty.
+   */
+  @Test
+  public void testReadRangeFilterEmptyResult() throws Exception {
+    writeTableWithIncrementalRecords();
+    Table table = tables.load(tableLocation);
+    LocalDateTime beginTime = LocalDateTime.of(2021, 1, 1, 0, 0, 0);
+    LocalDateTime endTime = LocalDateTime.of(2021, 2, 1, 0, 0, 0);
+    TableScan scan = table.newScan()
+            .filter(Expressions.and(
+                    Expressions.greaterThanOrEqual("timestamp", timestampToMicros(beginTime)),
+                    Expressions.lessThan("timestamp", timestampToMicros(endTime))));
+    int numRoots = 0;
+    for (ArrowBatch batch : new VectorizedTableScanIterable(scan, NUM_ROWS_PER_MONTH, false)) {
+      numRoots++;
+    }
+    assertEquals(0, numRoots);
+  }
+
+  /**
+   * Read all rows and selected columns from the table with a column selection filter. The test asserts that the Arrow
+   * {@link VectorSchemaRoot} contains the expected schema and expected vector types. Then the test asserts that the
+   * vectors contains expected values. The test also asserts the total number of rows match the expected value.
+   */
+  @Test
+  public void testReadColumnFilter1() throws Exception {
+    writeTableWithIncrementalRecords();
+    Table table = tables.load(tableLocation);
+    TableScan scan = table.newScan()
+        .select("timestamp", "int", "string");
+    readAndCheckVectorSchemaRoots(
+        scan, NUM_ROWS_PER_MONTH, 12 * NUM_ROWS_PER_MONTH,
+        ImmutableList.of("timestamp", "int", "string"));
+  }
+
+  /**
+   * Read all rows and a single column from the table with a column selection filter. The test asserts that the Arrow
+   * {@link VectorSchemaRoot} contains the expected schema and expected vector types. Then the test asserts that the
+   * vectors contains expected values. The test also asserts the total number of rows match the expected value.
+   */
+  @Test
+  public void testReadColumnFilter2() throws Exception {
+    writeTableWithIncrementalRecords();
+    Table table = tables.load(tableLocation);
+    TableScan scan = table.newScan()
+        .select("timestamp");
+    readAndCheckVectorSchemaRoots(
+        scan, NUM_ROWS_PER_MONTH, 12 * NUM_ROWS_PER_MONTH,
+        ImmutableList.of("timestamp"));
+  }
+
+  private void readAndCheckVectorSchemaRoots(
+      TableScan scan,
+      int numRowsPerRoot,
+      int expectedTotalRows,
+      List<String> columns) {
+    Set<String> columnSet = ImmutableSet.copyOf(columns);
+    int rowIndex = 0;
+    int totalRows = 0;
+    for (ArrowBatch batch : new VectorizedTableScanIterable(scan, numRowsPerRoot, false)) {
+      VectorSchemaRoot root = batch.createVectorSchemaRootFromVectors();
```

Comment:
As this is likely to be used as a reference for anyone looking to utilize the vectorized reader it would be good to include some tests of the `ArrowBatch` as well

Reply:
added tests for ColumnBatch